### PR TITLE
[Merged by Bors] - feat: port LinearAlgebra.QuadraticForm.Isometry

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/QuadraticForm/Isometry.lean [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+variable {Q₁ : QuadraticForm R M₁} {Q₂ : QuadraticForm R M₂} {Q₃ : QuadraticForm R M₃}
+
+-- Porting note: was `Coe`
+instance : CoeOut (Q₁.Isometry Q₂) (M₁ ≃ₗ[R] M₂) :=
+  ⟨Isometry.toLinearEquiv⟩
+
+-- Porting note: syntaut
+#noalign quadratic_form.isometry.to_linear_equiv_eq_coe
+
+instance : CoeFun (Q₁.Isometry Q₂) fun _ => M₁ → M₂ :=
```
Comment (ChrisHughes24) [inline Mathlib/LinearAlgebra/QuadraticForm/Isometry.lean:63]: We've been turning most of these into `EquivLike` or `FunLike` during the port.

Comment (Ruben-VandeVelde) [reply Mathlib/LinearAlgebra/QuadraticForm/Isometry.lean:63]: If you'd like to block this PR on that, can you explain what changes you need, exactly, or push them yourself?

Comment (ChrisHughes24) [reply Mathlib/LinearAlgebra/QuadraticForm/Isometry.lean]: I replaced the `CoeFun` with an `EquivLike` and restored `coe_toLinearEquiv`. These things have caused issues in the past, and it was a bit of work to fix all the breakages in #3082 for example

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**